liferay-frontend/liferay-portal
# LPS-131093 Add support for ERC to TaxonomyCategories

Related ticket -> [LPS-131093](https://issues.liferay.com/browse/LPS-131093)

This PR contains the implementation of the ERC for the entity `AssetCategory` scoped by group and the endpoints to CREATE, DELETE, and UPSERT categories.
- I had to include the property `siteId` in the TaxonomyCategory schema.
- I remove some permissions check in the PUT from the headless layer and change the local service used for the remote service to check the permissions.
- As a Category needs a Vocabulary, I've create the endpoint `/taxonomy-vocabularies/{taxonomyVocabularyId}/taxonomy-categories/by-externalReferenceCode/{externalReferenceCode}` for the ERC.
- Also, I've added the necessary methods to the services.

**Test Plan**
- Fetch this branch
- Deploy `portal-kernel` and `portal-impl`
- Do a DB upgrade
- Deploy `headless-admin-taxonomy-api` and `headless-admin-taxonomy-imp`
- Check that the endpoints works to create, delete, get and update asset categories
